uaclient/apt.py

```diff
@@ -453,6 +453,24 @@ def get_installed_packages_by_origin(origin: str) -> List[apt_pkg.Package]:
     return list(result)
 
 
+def get_alternative_versions_for_package(
+    package: apt_pkg.Package, exclude_origin: Optional[str] = None
+) -> List[apt_pkg.Version]:
+    alternative_versions = [
+        version
+        for version in package.version_list
+        if version != package.current_ver
+    ]
+    if exclude_origin:
+        for version in alternative_versions:
+            for file, _ in version.file_list:
+                if file.origin == exclude_origin:
+                    alternative_versions.remove(version)
+                    break
```

Comment:
Can't we already perform this logic when creating the `alternative_versions` variable ?

Reply:
Well, we could, but sincerely I don't see how that would simplify the logic. Do you have a suggestion?